src/test/java/io/confluent/connect/jdbc/sink/JdbcSinkTaskTest.java

```diff
@@ -291,4 +296,43 @@ void initWriter() {
     verifyAll();
   }
 
+  @Test
+  public void errorReporting() throws SQLException {
+    final int maxRetries = 2;
+    final int retryBackoffMs = 1000;
+
+    Set<SinkRecord> records = Collections.singleton(new SinkRecord("stub", 0, null, null, null, null, 0));
+    final JdbcDbWriter mockWriter = createMock(JdbcDbWriter.class);
+    SinkTaskContext ctx = createMock(SinkTaskContext.class);
+
+    mockWriter.write(records);
+    SQLException chainedException = new SQLException("cause 1");
+    chainedException.setNextException(new SQLException("cause 2"));
+    chainedException.setNextException(new SQLException("cause 3"));
+    expectLastCall().andThrow(chainedException).times(1);
+    mockWriter.write(anyObject());
+    expectLastCall();
+
+    JdbcSinkTask task = new JdbcSinkTask() {
+      @Override
+      void initWriter() {
+        this.writer = mockWriter;
+      }
+    };
+    task.initialize(ctx);
+    ErrantRecordReporter reporter = createMock(ErrantRecordReporter.class);
```

Comment:
Maybe also a testcase where both calls to `wirter.write()` failed and the reporter is called to send to the DLQ?

Reply:
Is the reporter not sending it to the DLQ in this test case?

Reply:
@aakashnshah  I don't think so. It looks like this testcase is setting up the `mockWriter.write` to throw the 1st time it is called, but not the subsequent calls for the individual retries. So, it would not go down the reporter path. If it did, you would probably see an exception about `reporter.report` call not expected since the testcases does not set up the expectation on mock.

Reply:
Ah I see what you are saying, thanks, I'll add that!